src/Control/Comonad/Cofree.hs

```diff
@@ -312,6 +315,17 @@ instance Traversable1 f => Traversable1 (Cofree f) where
     go (a :< as) = (:<) <$> f a <.> traverse1 go as
   {-# INLINE traverse1 #-}
 
+instance FunctorWithIndex i f => FunctorWithIndex [i] (Cofree f) where
+  imap f (a :< as) = f [] a :< imap (\i -> imap (f . (:) i)) as
+  {-# INLINE imap #-}
+
+instance FoldableWithIndex i f => FoldableWithIndex [i] (Cofree f) where
+  ifoldMap f (a :< as) = f [] a `mappend` ifoldMap (\i -> ifoldMap (f . (:) i)) as
+  {-# INLINE ifoldMap #-}
+
+instance TraversableWithIndex i f => TraversableWithIndex [i] (Cofree f) where
+  itraverse f (a :< as) = (:<) <$> f [] a <*> itraverse (\i -> itraverse (f . (:) i)) as
+  {-# INLINE itraverse #-}
```

Comment:
Currently, this is a bit cluttered:

https://github.com/ekmett/free/blob/7ca2d50f1a223a7eb628d0224a7cbc45916a8865/src/Control/Comonad/Cofree.hs#L326-L334

Can you add a newline between lines 329 and 330?